cmd/globals.go

```diff
@@ -200,5 +203,18 @@ func setGlobalsFromContext(ctx *cli.Context) error {
 			globalResolvers[host] = addr
 		}
 	}
+
+	customHeaders := ctx.StringSlice("custom-header")
+	if len(customHeaders) > 0 {
+		globalCustomHeader = make(http.Header)
+		for _, header := range customHeaders {
+			i := strings.IndexByte(header, ':')
+			if i <= 0 {
+				return fmt.Errorf("invalid custom header entry %s", header)
+			}
+			globalCustomHeader.Add(header[:i], header[i+1:])
```

Comment:
Do we need to trim the value? It's very common to specify `X-Test: 123`:
```suggestion
			globalCustomHeader.Add(header[:i], strings.TrimSpace(header[i+1:]))
```
Now this causes this to add the `X-Test` header to be specified with the value ` 123` (incl. leading space). Although mostly not an issue, this may be unexpected.

Reply:
Trimspace on both should be fine. This is not performance critical, and leading/trailing whitespace doesn't make sense in either key nor value.

Reply:
Use httpguts https://pkg.go.dev/golang.org/x/net/http/httpguts if you want to be pedantic